### PR TITLE
Replace use of 2to3 with six

### PR DESCRIPTION
**Pull request by rahulporuri**
`modernize` was used to make the necessary changes.
The changes introduced in this PR are :

- remove use of `xrange`
- remove use of `basestring` and `unicode`
- update use of `exception` statement
- update use of `raise` statement
- remove/replace use of `print` statement
- make imports explcitly local
- update modules imports to be 2/3 compat e.g. `cStringIO`, `thread` and `cPickle`
- update dict-related methods i.e. `has_key`, `values`, `keys`, `items`
- update function object attrs e.g. `func_name`

### CODE REVIEW

**Comment by rahulporuri** [comment]
The test failures on Python seem to be related to qt-api preparation. See traceback below for python 2 and toolkit pyqt on rh6.

```
$ edm run -e pyface-test-2.7-pyqt -- python -m nose pyface/tasks/tests/test_dock_pane_toggle_group.py 
E
======================================================================
ERROR: Failure: ValueError (API 'QDate' has already been set to version 1. Pyface expects PyQt API 2 under Python 2. Either import Pyface before any other Qt-using packages, or explicitly set the API before importing any other Qt-using packages.)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/rporuri/.edm/envs/pyface-test-2.7-pyqt/lib/python2.7/site-packages/nose/loader.py", line 420, in loadTestsFromName
    addr.filename, addr.module)
  File "/home/rporuri/.edm/envs/pyface-test-2.7-pyqt/lib/python2.7/site-packages/nose/importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/home/rporuri/.edm/envs/pyface-test-2.7-pyqt/lib/python2.7/site-packages/nose/importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/home/rporuri/Github/work/pyface/pyface/tasks/tests/test_dock_pane_toggle_group.py", line 5, in <module>
    from pyface.tasks.action.api import SMenu, SMenuBar, SGroup, \
  File "/home/rporuri/Github/work/pyface/pyface/tasks/action/api.py", line 13, in <module>
    from .dock_pane_toggle_group import DockPaneToggleGroup
  File "/home/rporuri/Github/work/pyface/pyface/tasks/action/dock_pane_toggle_group.py", line 5, in <module>
    from pyface.action.api import Action, ActionItem, Group
  File "/home/rporuri/Github/work/pyface/pyface/action/api.py", line 15, in <module>
    from .action_item import ActionItem
  File "/home/rporuri/Github/work/pyface/pyface/action/action_item.py", line 25, in <module>
    from pyface.toolkit import toolkit_object
  File "/home/rporuri/Github/work/pyface/pyface/toolkit.py", line 27, in <module>
    toolkit = toolkit_object = find_toolkit('pyface.toolkits')
  File "/home/rporuri/Github/work/pyface/pyface/base_toolkit.py", line 272, in find_toolkit
    toolkit = plugin.load()
  File "/home/rporuri/.edm/envs/pyface-test-2.7-pyqt/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2408, in load
    return self.resolve()
  File "/home/rporuri/.edm/envs/pyface-test-2.7-pyqt/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2414, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "/home/rporuri/Github/work/pyface/pyface/ui/qt4/init.py", line 18, in <module>
    from pyface.qt import QtCore, QtGui, qt_api
  File "/home/rporuri/Github/work/pyface/pyface/qt/__init__.py", line 82, in <module>
    prepare_pyqt4()
  File "/home/rporuri/Github/work/pyface/pyface/qt/__init__.py", line 44, in prepare_pyqt4
    raise ValueError(msg)
ValueError: API 'QDate' has already been set to version 1. Pyface expects PyQt API 2 under Python 2. Either import Pyface before any other Qt-using packages, or explicitly set the API before importing any other Qt-using packages.

----------------------------------------------------------------------
```

**Comment by corranwebster** [comment]
Thanks for doing this - it has been on my to-do list for a while but have not gotten around to it.

File: pyface/timer/tests/test_timer.py [hunk 1]
```diff
@@ -101,7 +102,7 @@ def test_interval(self):
             expected <= actual
             for expected, actual in zip(expected_times, handler.times)
         ):
-            print(handler.times)
+            print((handler.times))
```
Comment (corranwebster) [inline pyface/timer/tests/test_timer.py]: Extra parenthesis.

File: pyface/timer/tests/test_timer.py [hunk 2]
```diff
@@ -128,7 +129,7 @@ def test_expire(self):
         if not all(
             t < timer._start_time + timer.expire + 0.01 for t in handler.times
         ):
-            print(handler.times[-1], timer._start_time + timer.expire)
+            print((handler.times[-1], timer._start_time + timer.expire))
```
Comment (corranwebster) [inline pyface/timer/tests/test_timer.py]: Extra parenthesis.

File: pyface/timer/tests/test_timer.py [hunk 3]
```diff
@@ -212,7 +213,7 @@ def test_interval(self):
             expected <= actual
             for expected, actual in zip(expected_times, handler.times)
         ):
-            print(handler.times)
+            print((handler.times))
```
Comment (corranwebster) [inline pyface/timer/tests/test_timer.py]: Extra parenthesis.

File: pyface/timer/tests/test_timer.py [hunk 4]
```diff
@@ -240,7 +241,7 @@ def test_expire(self):
         if not all(
             t < timer._start_time + timer.expire + 0.01 for t in handler.times
         ):
-            print(handler.times[-1], timer._start_time + timer.expire)
+            print((handler.times[-1], timer._start_time + timer.expire))
```
Comment (corranwebster) [inline pyface/timer/tests/test_timer.py]: Extra parenthesis.

File: pyface/ui/qt4/console/console_widget.py [hunk 5]
```diff
@@ -104,7 +104,7 @@ class ConsoleWidget(QtGui.QWidget):
 
     # The shortcuts defined by this widget. We need to keep track of these to
     # support 'override_shortcuts' above.
-    _shortcuts = set(_ctrl_down_remap.keys() +
+    _shortcuts = set(list(_ctrl_down_remap.keys()) +
```
Comment (corranwebster) [inline pyface/ui/qt4/console/console_widget.py]: This line can be re-written more cleanly using sets directly.

**Pull request merged**